modules/basicfuncs/tests/test_basicfuncs.c

```diff
@@ -184,6 +184,15 @@ Test(basicfuncs, test_str_funcs)
   assert_template_format("$(binary 1 0x1)", "\1\1");
   assert_template_format("$(binary 0xFF 255 0377)", "\xFF\xFF\xFF");
   assert_template_format_with_len("$(binary 0xFF 0x00 0x40)", "\xFF\000@", 3);
+
+  assert_template_format("[$(base64-encode abc)]", "[YWJj]");
```

Comment:
I am missing a test case where argc > 1 (this is why we have a loop inside the template function), because some might accidentally expect to get their parameters encoded one by one.
 - encode(apple) + encode(apple) `!=` encode(appleapple)

while
 - decode( encode(apple) + encode(apple) ) `==` decode (encode(appleapple) )

note: This only needs to be considered in cases where someone might want to handle the raw output of the encoding.


I also wrote two simple tests, to raise the attention to use proper escaping:
`assert_template_format("[$(base64-encode 'X  X')]", "[WCAgWA==]");`
`assert_template_format("[$(base64-encode X  X)]", "[WFg=]");`
`// $(base64-encode X  X) == $(base64-encode 'X'   'X') == $(base64-encode 'XX')`

Reply:
Good point. I have added a testcase for the loop, thanks for the heads up.